# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd

In [2]:
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [5]:
series_1 = pd.Series(np.random.randint(1, 5, size = 100))
series_2 = pd.Series(np.random.randint(1, 4, size = 100))
series_3 = pd.Series(np.random.randint(10000, 30001, size = 100))


In [7]:
series_1[0:10]

0    1
1    2
2    3
3    3
4    2
5    1
6    4
7    3
8    3
9    4
dtype: int32

In [8]:
series_2[0:10]

0    2
1    2
2    2
3    1
4    3
5    3
6    1
7    3
8    1
9    1
dtype: int32

In [9]:
series_3[0:10]

0    24164
1    15232
2    12506
3    13790
4    26300
5    17924
6    20548
7    15002
8    10278
9    12834
dtype: int32

### Step 3. Let's create a DataFrame by joinning the Series by column

In [27]:
frame = { 'series_1': series_1, 'series_2': series_2, 'series_3': series_3 }

In [31]:
df = pd.DataFrame(frame)

In [34]:
df.head()

series_1  series_2  series_3
0         1         2     24164
1         2         2     15232
2         3         2     12506
3         3         1     13790
4         2         3     26300

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [36]:
df.columns = ['bedrs', 'bathrs', 'price_sqr_meter']

In [37]:
df

bedrs  bathrs  price_sqr_meter
0       1       2            24164
1       2       2            15232
2       3       2            12506
3       3       1            13790
4       2       3            26300
..    ...     ...              ...
95      4       3            11947
96      4       2            16162
97      2       2            23666
98      1       3            12591
99      1       3            28404

[100 rows x 3 columns]

In [43]:
df.rename(columns = {0: 'bedrs', 1: 'bathrs', 2: 'price_sqr_meter'}, inplace=True)

df.head()


bedrs  bathrs  price_sqr_meter
0      1       2            24164
1      2       2            15232
2      3       2            12506
3      3       1            13790
4      2       3            26300

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [56]:
bigcolumn = pd.DataFrame(pd.concat([series_1, series_2, series_3]))

In [57]:
bigcolumn.head()

0
0  1
1  2
2  3
3  3
4  2

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [58]:
bigcolumn.shape[0]

300

In [59]:
bigcolumn.reset_index(inplace=True)

In [60]:
bigcolumn

index      0
0        0      1
1        1      2
2        2      3
3        3      3
4        4      2
..     ...    ...
295     95  11947
296     96  16162
297     97  23666
298     98  12591
299     99  28404

[300 rows x 2 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [61]:
bigcolumn.reset_index(inplace=True)

In [62]:
bigcolumn

level_0  index      0
0          0      0      1
1          1      1      2
2          2      2      3
3          3      3      3
4          4      4      2
..       ...    ...    ...
295      295     95  11947
296      296     96  16162
297      297     97  23666
298      298     98  12591
299      299     99  28404

[300 rows x 3 columns]